### DISCUSSION TITLE
Experimental feature "dynamicIO" only available on Canary

### DISCUSSION THREAD

**Discussion by emulienfou** [vercel/next.js#72101] [Help]
### Summary

According to the current stable documentation (Next 15.0.2) the page for `unstable_cache`: https://nextjs.org/docs/app/api-reference/legacy-apis/unstable_cache says: `In version 15, we recommend using the "use cache" directive instead.`

**Problem:**

Now, according to the documentation here: https://nextjs.org/docs/app/api-reference/directives/use-cache
A developer need to enable the experimental feature `dynamicIO`

However, when doing this with stable version `15.0.2` the server return the next error:
```
CanaryOnlyError: The experimental feature "experimental.dynamicIO" can only be enabled when using the latest canary version of Next.js.
```

**Change**

- Either the documentation is missing that the feature is only available for Canary only
- Or, the documentation needs to be updated to says that we can still use `unstable_cache`
- Or, the `use cache` needs to be made available for version 15.x like the current documentation says today

**Comment by icyJoseph** [comment]
Good suggestion. For what is worth, the blog post introducing this, https://nextjs.org/blog/our-journey-with-caching#experimental, does say that it is only in canary.

**Comment by tpiros** [comment]
I agree with @emulienfou - it'd be great to have some clarify on this. Should we use `unstable_cache` in Next.js 15, then? I would like to have caching enabled for a production app and as such it is not recommended for me to use `use cache` (based on the post listed by @icyJoseph). So what to do? :)

**Comment by alexanderPastornicky** [comment]
I have upgraded to version 15.0.3 (latest), but I still get this error.

**Reply by icyJoseph** [reply]
But, isn't it `next@canary` that one should have?

**Reply by JwillSDP** [reply]
I get the error too

**Comment by abkorim1998** [comment]
i got same issue.
but i fixed it by installing  `next@canary` exmple `npm install next@canary`
here is the sorce: https://nextjs.org/docs/messages/ppr-preview

**Comment by vladkampov** [comment]
I wonder if there is any news on this one. Would it make it to the stable version?
